cmd/nerdctl/ps.go

```diff
@@ -319,3 +385,355 @@ func getContainerSize(ctx context.Context, client *containerd.Client, c containe
+			status := containerd.ProcessStatus(value)
+			switch status {
+			case containerd.Running, containerd.Created, containerd.Stopped, containerd.Paused, containerd.Pausing, containerd.Unknown:
+				statusFilterFuncs = append(statusFilterFuncs, func(stats containerd.ProcessStatus) bool {
+					return status == stats
+				})
+			case containerd.ProcessStatus("exited"):
+				statusFilterFuncs = append(statusFilterFuncs, func(stats containerd.ProcessStatus) bool {
+					return containerd.Stopped == stats
+				})
+			case containerd.ProcessStatus("restarting"), containerd.ProcessStatus("removing"), containerd.ProcessStatus("dead"):
+				logrus.Warnf("%s is not supported and is ignored", filter)
+				return nil
+			default:
+				return fmt.Errorf("invalid filter '%s'", filter)
+			}
+			return nil
+		}},
+		{"before", func(filter, value string) error {
+			beforeC, err := idOrNameFilter(ctx, conts, value)
+			if err == nil {
+				beforeFilterFuncs = append(beforeFilterFuncs, func(t time.Time) bool {
+					return t.Before(beforeC.CreatedAt)
+				})
+			}
+			return err
+		}},
+		{"since", func(filter, value string) error {
+			sinceC, err := idOrNameFilter(ctx, conts, value)
+			if err == nil {
+				sinceFilterFuncs = append(sinceFilterFuncs, func(t time.Time) bool {
+					return t.After(sinceC.CreatedAt)
+				})
+			}
+			return err
+		}},
+		{"id", func(filter, value string) error {
+			idFilterFuncs = append(idFilterFuncs, func(id string) bool {
+				if value == "" {
+					return false
+				}
+				return strings.HasPrefix(id, value)
+			})
+			return nil
+		}},
+		{"name", func(filter, value string) error {
+			nameFilterFuncs = append(nameFilterFuncs, func(name string) bool {
+				if value == "" {
+					return true
+				}
+				return strings.Contains(name, value)
+			})
+			return nil
+		}},
+		{"label", func(filter, value string) error {
+			k, v, hasValue := value, "", false
+			if subs := strings.SplitN(value, "=", 2); len(subs) == 2 {
+				hasValue = true
+				k, v = subs[0], subs[1]
+			}
+			labelFilterFuncs = append(labelFilterFuncs, func(labels map[string]string) bool {
+				if labels == nil {
+					return false
+				}
+				val, ok := labels[k]
+				if !ok || (hasValue && val != v) {
+					return false
+				}
+				return true
+			})
+			return nil
+		}},
+		{"volume", func(filter, value string) error {
+			volumeFilterFuncs = append(volumeFilterFuncs, func(vols []*containerVolume) bool {
+				for _, vol := range vols {
+					if (vol.Source != "" && vol.Source == value) ||
+						(vol.Destination != "" && vol.Destination == value) ||
+						(vol.Name != "" && vol.Name == value) {
+						return true
+					}
+				}
+				return false
+			})
+			return nil
+		}},
+		{"network", func(filter, value string) error {
+			networkFilterFuncs = append(networkFilterFuncs, func(networks []string) bool {
+				for _, network := range networks {
+					if network == value {
+						return true
+					}
+				}
+				return false
+			})
+			return nil
+		}},
+	}
```

Comment:
I think refactor these codes to a single function name like `getFilterFuncs` may be a good idea.

It will make the codebase more readable